builder_test.go

```diff
@@ -84,553 +93,348 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
+					{Name: "some-unmet-dep", Version: "v2"},
+					{Name: "other-dep", Version: "v4"},
+				}}
+				bpB.EXPECT().Build(expectedPlanB, config)
 
-				mkfile(t,
-					"[[entries]]\n"+
-						`name = "dep1"`+"\n"+
-						`version = "v1"`+"\n"+
-						"[[entries]]\n"+
-						`name = "dep1-replace"`+"\n"+
-						`version = "v7"`+"\n",
-					filepath.Join(appDir, "build-plan-out-A-v1.toml"),
-				)
-				mkfile(t,
-					"[[entries]]\n"+
-						`name = "dep1-next"`+"\n"+
-						`version = "v9"`+"\n"+
-						"[[entries]]\n"+
-						`name = "dep2"`+"\n"+
-						`version = "v4"`+"\n"+
-						"[[entries]]\n"+
-						`name = "dep2-replace"`+"\n"+
-						`version = "v8"`+"\n",
-					filepath.Join(appDir, "build-plan-out-B-v2.toml"),
-				)
-				metadata, err := builder.Build()
+				_, err := builder.Build()
 				if err != nil {
 					t.Fatalf("Unexpected error:\n%s\n", err)
 				}
-				if s := cmp.Diff(metadata, &lifecycle.BuildMetadata{
-					Processes: []launch.Process{},
-					Buildpacks: []lifecycle.Buildpack{
-						{ID: "A", Version: "v1", API: "0.5", Homepage: "Buildpack A Homepage"},
-						{ID: "B", Version: "v2", API: "0.2"},
-					},
-					BOM: []lifecycle.BOMEntry{
-						{
-							Require:   lifecycle.Require{Name: "dep1", Version: "v1"},
-							Buildpack: lifecycle.Buildpack{ID: "A", Version: "v1"},
-						},
-						{
-							Require:   lifecycle.Require{Name: "dep1-replace", Version: "v7"},
-							Buildpack: lifecycle.Buildpack{ID: "A", Version: "v1"},
-						},
-						{
-							Require:   lifecycle.Require{Name: "dep1-next", Version: "v9"},
-							Buildpack: lifecycle.Buildpack{ID: "B", Version: "v2"},
-						},
-						{
-							Require:   lifecycle.Require{Name: "dep2", Version: "v4"},
-							Buildpack: lifecycle.Buildpack{ID: "B", Version: "v2"},
-						},
-						{
-							Require:   lifecycle.Require{Name: "dep2-replace", Version: "v8"},
-							Buildpack: lifecycle.Buildpack{ID: "B", Version: "v2"},
-						},
-					},
-				}); s != "" {
-					t.Fatalf("Unexpected:\n%s\n", s)
-				}
-
-				testPlan(t,
-					[]lifecycle.Require{
-						{Name: "dep1", Version: "v1"},
-						{Name: "dep1-next", Version: "v2"},
-						{Name: "dep1-replace", Version: "v3"},
-					},
-					filepath.Join(appDir, "build-plan-in-A-v1.toml"),
-				)
-
-				testPlan(t,
-					[]lifecycle.Require{
-						{Name: "dep1-next", Version: "v2"},
-						{Name: "dep2", Version: "v4"},
-						{Name: "dep2-next", Version: "v5"},
-						{Name: "dep2-replace", Version: "v6"},
-					},
-					filepath.Join(appDir, "build-plan-in-B-v2.toml"),
-				)
 			})
 
-			it("should convert metadata version to top level version for buildpacks with buildpack api 0.2", func() {
-				builder.Plan = lifecycle.BuildPlan{
-					Entries: []lifecycle.BuildPlanEntry{
-						{
-							Providers: []lifecycle.Buildpack{
-								{ID: "B", Version: "v2"},
+			when("build metadata", func() {
+				when("bom", func() {
+					it("should convert top level version to metadata.version", func() {
```

Comment:
Should we need to do a conversion in the default case here? It might be nice if the buildpack always returned the data in a consistent way regardless of the bp API (never top-level) and the the builder only converted provide labels in the legacy format.